File: examples/end2end/seq_gen.js
// Sequence generator
/* jshint node: true */

(function () {
    "use strict";

    var Topic = require("./out/franz").Topic;

    var t = new Topic("http://127.0.0.1:8080");
    var curValues = new Map();
    var finValues = new Map();

    function doit() {
        curValues.forEach(function (curValue, tag) {
            t.push(tag + ":" + curValue.toString());
            curValue += 1;
            if (curValue === finValues.get(tag)) {
                curValues.delete(tag);
                finValues.delete(tag);
                t.push(tag + ":done");
            } else {
                curValues.set(tag, curValue);
            }
        });

        var pieces = t.pop().split(":");
        if (pieces[0] !== "sequence") { return; }
        var count = parseInt(pieces[1]);
        var tag = pieces[2];

        curValues.set(tag, 0);
        finValues.set(tag, count);
    }

    setInterval(doit, 300);
})();

File: examples/end2end/seq_gen.py
# Squares stuff

import sys
sys.path.append("./out")

from franz import Queue


def main():
    q = Queue("http://127.0.0.1:8080/simple")
    tracked = {}
    while True:
        pieces = q.pop().split(":")
        print "popped", pieces
        tag = pieces[0]
        if tag in tracked:
            value = pieces[1]
            if value == "done":
                tracked.pop(tag)
            else:
                value = int(value)
                if value < tracked[tag]:
                    q.push(tag + ":" + str(value+1))
                else:
                    q.push(tag + ":done")
        elif tag == "sequence":
            tag = pieces[2]
            tracked[tag] = int(pieces[1])
            q.push(tag + ":0")


if __name__ == '__main__':
    main()
